### DISCUSSION TITLE
Rules on button press and release

### DISCUSSION THREAD

**Discussion by avishorp** [arendst/Tasmota#17460] [Q&A] [answered]
I want to control a Shelly 2.5 (stock firmware) with a Tasmota flashed Sonoff touch switch. I need a way to publish an MQTT message when a button it pressed (touched) and another message when it is released. Is it possible to implement such functionality with rules?

**Comment by barbudor** [comment, marked as answer]
Not with a Button. A button will only provide the event Toggle when depressed 

You need to configure as a Switch and then apply the Switchmode 1 or 2 depending on the cabling of your switch and if you want on/off at press/depress or the opposite

The rule would then looks like
```
Rule1
    On Switch1#state==0 do Publish yourshelly/sometopic somevalue endon
    On Switch1#state==1 do Publish yourshelly/sometopic somevalue endon
```

**Comment by avishorp** [comment]
The problem here is that SwitchMode is global. In my case, I want two buttons to move a shutter up and down (active when held) and one button to act as a light switch (touch to toggle). Setting SwitchMode to 1/2 would prevent the light button from working.

**Reply by sfromis** [reply]
No [SwitchMode](https://tasmota.github.io/docs/Commands/#switchmode) is not global, you can set it per pin configured as Switch.

**Reply by avishorp** [reply]
The documentation does not mention that. I tried adding the switch number after SwitchMode command in the console, but it does not seem to have any effect.

**Reply by barbudor** [reply]
From: https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode
> If there is more than one Switch<x> component, use `SwitchMode<x>` where `<x>` is the number of your switch from the Tasmota GPIO configuration.

Generally it makes it easier for us to help if you provide exact command typed and a copy/paste of the logs from the console

**Reply by sfromis** [reply]
.... because "after" does not tell what exactly you did, like possibly inserting spaces where they should not be ... nor what you mean by "not seem".

**Reply by avishorp** [reply]
You're right, I missed that paragraph from the docs. Thanks!
